batch/batch/driver/instance_collection.py

```diff
@@ -154,12 +154,16 @@ async def check_on_instance(self, instance):
 
         if gce_state in ('STAGING', 'RUNNING'):
             last_start_timestamp = spec.get('lastStartTimestamp')
-            assert last_start_timestamp is not None, f'lastStartTimestamp does not exist {spec}'
-            last_start_time_msecs = dateutil.parser.isoparse(last_start_timestamp).timestamp() * 1000
-
-            if instance.state == 'pending' and time_msecs() - last_start_time_msecs > 5 * 60 * 1000:
-                log.exception(f'{instance} did not activate within 5m after starting, deleting')
-                await self.call_delete_instance(instance, 'activation_timeout')
+            if last_start_timestamp is not None:
+                last_start_time_msecs = dateutil.parser.isoparse(last_start_timestamp).timestamp() * 1000
+
+                if instance.state == 'pending' and time_msecs() - last_start_time_msecs > 5 * 60 * 1000:
+                    log.exception(f'{instance} did not activate within 5m after starting, deleting')
+                    await self.call_delete_instance(instance, 'activation_timeout')
+            else:
+                time_elapsed = time_msecs() - instance.time_created
+                if instance.state == 'pending' and time_elapsed > 5 * 60 * 1000:
```

Comment:
nit: the use of a `time_elapsed` variable is inconsistent between the two `if`s that are otherwise the same. I like having the variable